src/neo/Wallets/Wallet.cs

```diff
@@ -540,6 +541,7 @@ public long CalculateNetworkFee(DataCache snapshot, Transaction tx)
                         if (witness.ScriptHash == hash)
                         {
                             witness_script = witness.VerificationScript;
+                            invocationScript = witness.InvocationScript;
```

Comment:
Unfortunately, this won't work for the contract-based witnesses, because such witnesses have empty verification script => condition `(witness.ScriptHash == hash)` is false (it is located 3 lines above). And we need invocation script only for contract-based witnesses.

Reply:
Please take a look again